### PR TITLE
cody: configurable match limit for global policies

### PR DESCRIPTION
**Pull request by stefanhengl**
If a user creates a global policy (a policy without a repo pattern), the policy matches up to embeddings.policyRepositoryMatchLimit repos (default=5000).

Previously the limit was hard coded to 5000.

I chose the name to match Code intel's "codeIntelAutoIndexing.policyRepositoryMatchLimit".

## Test plan:
- new unit tests
- Set various values for embeddings.PolicyRepositoryMatchLimit in site-config, create a global policy, and verify that jobs created in site-admin > embeddings jobs matches the expectation.

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -2395,6 +2395,14 @@
           "description": "The time to wait between runs. Valid time units are \"s\", \"m\", \"h\". Example values: \"30s\", \"5m\", \"1h\".",
           "type": "string",
           "default": "24h"
+        },
+        "policyRepositoryMatchLimit": {
```
Comment (jtibshirani) [inline schema/site.schema.json:2399]: The list of embeddings settings is getting quite long, maybe it'd be nice to nest the new policy settings in their own subsection.

File: enterprise/internal/embeddings/background/repo/store.go [hunk 2]
```diff
@@ -213,25 +218,46 @@ func embeddingConfigValidator(q conftypes.SiteConfigQuerier) conf.Problems {
...
 	minimumIntervalString := embeddingsConf.MinimumInterval
 	d, err := time.ParseDuration(minimumIntervalString)
-	if err != nil {
-		return EmbeddableRepoOpts{MinimumInterval: defaultMinimumInterval}
+	if err == nil {
+		opts.MinimumInterval = d
 	}
 
-	return EmbeddableRepoOpts{MinimumInterval: d}
+	opts.PolicyRepositoryMatchLimit = embeddingsConf.PolicyRepositoryMatchLimit
```
Comment (jtibshirani) [inline enterprise/internal/embeddings/background/repo/store.go:242]: If the embeddings config is non-nil, but we don't define a policy match limit, then we don't apply the default of 5000 and the match is unlimited. This is probably unintentional?

Comment (stefanhengl) [reply enterprise/internal/embeddings/background/repo/store.go]: Oops! Thanks for catching this. I updated the test accordinlgy.

File: schema/site.schema.json (hunk 1 above)
Comment (stefanhengl) [reply schema/site.schema.json:2399]: I prefer flat lists (even if they are a bit longer) over nesting but I don't mind changing it if you think it helps admins. We currently only have 2 settings that are related to policies (minimumInterval and policyRepositoryMatchLimit). Do you want to nest them under embeddings.policies?

Comment (jtibshirani) [reply schema/site.schema.json:2399]: I don't feel strongly about it at all, we can leave as-is.

File: enterprise/internal/embeddings/background/repo/store.go [hunk 3]
```diff
@@ -213,25 +218,48 @@ func embeddingConfigValidator(q conftypes.SiteConfigQuerier) conf.Problems {
...
 	d, err := time.ParseDuration(minimumIntervalString)
-	if err != nil {
-		return EmbeddableRepoOpts{MinimumInterval: defaultMinimumInterval}
+	if err == nil {
+		opts.MinimumInterval = d
 	}
 
-	return EmbeddableRepoOpts{MinimumInterval: d}
+	if embeddingsConf.PolicyRepositoryMatchLimit != nil {
+		opts.PolicyRepositoryMatchLimit = embeddingsConf.PolicyRepositoryMatchLimit
```
Comment (jtibshirani) [inline enterprise/internal/embeddings/background/repo/store.go:243]: Small comment, I don't think `EmbeddableRepoOpts.PolicyRepositoryMatchLimit` needs to be a pointer anymore. Now we just know it's disabled if the number is negative.

**Review by jtibshirani — APPROVED** [review]

File: enterprise/internal/embeddings/background/repo/store.go (hunk 3 above)
Comment (stefanhengl) [reply enterprise/internal/embeddings/background/repo/store.go:243]: I think we still need it to distinguish 0 from "unset". For "unset" we want to use the default, for 0 we want to actually use 0.  We have a couple of other settings for which 0 means default. EG for MaxCodeEmbeddingsPerRepo, 0 means 3072000, but to me that is very confusing.

**Pull request merged**